### PR TITLE
Internal: Editor V2 - Manage Website button [ED-9717]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: core/editor/config-providers/editor-v2-config-provider.php [hunk 1]
```diff
@@ -74,9 +74,25 @@ public function get_script_handles_to_enqueue() {
...
 
+	public function get_client_settings() {
+		$common_configs = Editor_Common_Configs::get_client_settings();
+
+		$v2_config = [
+			'handle' => 'elementor-editor-loader-v2',
+			'name' => 'elementorEditorV2Config',
+			'config' => [
+				'urls' => [
+					'dashboard' => admin_url(),
```
Comment (Nevoss) [inline core/editor/config-providers/editor-v2-config-provider.php:85]:
```suggestion
					'admin_url' => admin_url(),
```

File: core/editor/config-providers/editor-common-configs.php [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+					'google-font-roboto',
+					'flatpickr',
+					'pickr',
+				],
+			],
+		];
+	}
+
+	public static function get_client_settings() {
+		$suffix = ( Utils::is_script_debug() || Utils::is_elementor_tests() ) ? '' : '.min';
```
Comment (Nevoss) [inline core/editor/config-providers/editor-common-configs.php]: Move to class

File: core/editor/editor-loader.php [hunk 3]
```diff
@@ -52,6 +52,18 @@ public function register_scripts() {
 		}
 	}
 
+	public function print_client_settings() {
+		$client_settings = $this->config_provider->get_client_settings();
```
Comment (Nevoss) [inline core/editor/editor-loader.php:56]: Normalize and validations like assets

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: core/editor/assets/js/editor-loader-v2.js [hunk 4]
```diff
@@ -12,4 +12,7 @@ if ( ! window.__UNSTABLE__elementorPackages?.editor ) {
 
 window.__UNSTABLE__elementorPackages
     .editor
-    .init( document.getElementById( 'elementor-editor-wrapper-v2' ) );
+    .init(
+		document.getElementById( 'elementor-editor-wrapper-v2' ),
+		window.elementorEditorV2Config,
+	);
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply core/editor/assets/js/editor-loader-v2.js:18]: Done

File: core/editor/config-providers/editor-v2-config-provider.php (hunk 1 above)
Comment (StyleShit) [reply core/editor/config-providers/editor-v2-config-provider.php:85]: Done

File: core/editor/config-providers/editor-common-configs.php (hunk 2 above)
Comment (StyleShit) [reply core/editor/config-providers/editor-common-configs.php]: Done

File: core/editor/editor-loader.php (hunk 3 above)
Comment (StyleShit) [reply core/editor/editor-loader.php:56]: Done

File: tests/phpunit/elementor/core/editor/test-editor-loader.php [hunk 5]
```diff
@@ -130,6 +130,88 @@ public function test_load_script_translations() {
 		);
 	}
 
+	public function test_print_client_settings() {
+		// Arrange
+		$this->mock_config_provider->method( 'get_script_configs' )->willReturn( [
```
Comment (Nevoss) [inline tests/phpunit/elementor/core/editor/test-editor-loader.php:135]: try to use data_provider

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: core/editor/config-providers/editor-common-configs.php [hunk 6]
```diff
@@ -158,4 +158,14 @@ public static function get_style_configs() {
 			],
 		];
 	}
+
+	public static function get_client_settings() {
+		return [
+			[
+				'handle' => 'elementor-editor',
+				'name' => 'ElementorConfig',
+				'config' => Editor_Common_Client_Settings::get_client_settings(),
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply core/editor/config-providers/editor-common-configs.php:167]: Done

File: core/editor/config-providers/editor-v2-config-provider.php [hunk 7]
```diff
@@ -74,9 +74,25 @@ public function get_script_handles_to_enqueue() {
 			->values();
 	}
 
+	public function get_client_settings() {
+		$common_configs = Editor_Common_Configs::get_client_settings();
+
+		$v2_config = [
+			'handle' => 'elementor-editor-loader-v2',
+			'name' => 'elementorEditorV2Config',
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply core/editor/config-providers/editor-v2-config-provider.php:82]: Done

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		expect( result.current ).toEqual( {
+			urls: {
+				admin: 'https://localhost/wp-admin/',
+			},
+		} );
+	} );
+
+	it( 'should throw when not rendered inside a <SettingsContextProvider />', () => {
+		// Arrange.
+		// TODO: Mock the `console.error` because `@testing-library/react-hooks` takes over the `console.error`
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:26]: Done

File: packages/packages/editor/src/index.ts [hunk 9]
```diff
@@ -1,2 +1,3 @@
 export { injectIntoTop } from './locations';
 export { default as init } from './init';
+export { useSettings } from './contexts/settings-context';
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply packages/packages/editor/src/index.ts:3]: Done

File: tests/phpunit/elementor/core/editor/test-editor-loader.php (hunk 5 above)
Comment (StyleShit) [reply tests/phpunit/elementor/core/editor/test-editor-loader.php]: Done

**Review by Nevoss — DISMISSED** [review]

File: packages/packages/editor/src/init.tsx [hunk 10]
```diff
@@ -1,12 +1,22 @@
...
 
-export default function init( domElement: HTMLElement ): void {
+export default function init( domElement: HTMLElement, settings: Settings ): void {
 	const store = createStore();
 
 	dispatchReadyEvent();
 
-	ReactDOM.render( <StoreProvider store={ store }><Shell /></StoreProvider>, domElement );
+	ReactDOM.render( (
+		<SettingsContextProvider settings={ settings }>
```
Comment (rotemee) [inline packages/packages/editor/src/init.tsx:14]:
```suggestion
		<SettingsProvider settings={ settings }>
```

File: tests/phpunit/elementor/core/editor/test-editor-loader.php [hunk 11]
```diff
@@ -202,4 +291,6 @@ public function load_script_translations__with_replace_requested_file__data_prov
 			],
 		];
 	}
+
+
```
Comment (rotemee) [inline tests/phpunit/elementor/core/editor/test-editor-loader.php:295]: Remove.

File: tests/phpunit/elementor/core/editor/test-editor-loader.php [hunk 12]
```diff
@@ -130,6 +130,95 @@ public function test_load_script_translations() {
...
+		// Assert
+		$this->assertEquals(
+			$expected,
+			wp_scripts()->get_data( $client_settings['handle'] ?? null, 'before' )
+		);
+	}
+
+	public function print_client_settings_data_provider() {
+		return [
```
Comment (rotemee) [inline tests/phpunit/elementor/core/editor/test-editor-loader.php:165]: Explain by a comment above each case.

File: packages/packages/editor/src/contexts/settings-context.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+
+export interface Settings {
+	urls: {
+		admin: string,
+	},
+}
+
+const SettingsContext = createContext<Settings | null>( null );
+
+export function SettingsContextProvider( { children, settings }: PropsWithChildren<{ settings: Settings }> ) {
```
Comment (rotemee) [inline packages/packages/editor/src/contexts/settings-context.tsx:11]:
```suggestion
export function SettingsProvider( { children, settings }: PropsWithChildren<{ settings: Settings }> ) {
```

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+describe( '@elementor/editor - useSettings()', () => {
+	it( 'should return the value from the context', () => {
+		// Act.
+		const { result } = renderHookWithSettings(
+			() => useSettings(),
+			{
+				urls: {
+					admin: 'https://localhost/wp-admin/',
+				},
```
Comment (rotemee) [inline packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:12]: Move out?

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+		// Assert.
+		expect( result.current ).toEqual( {
+			urls: {
+				admin: 'https://localhost/wp-admin/',
+			},
+		} );
+	} );
+
+	it( 'should throw when not rendered inside a <SettingsContextProvider />', () => {
```
Comment (rotemee) [inline packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:24]:
```suggestion
	it( 'should throw when not rendered inside a <SettingsProvider />', () => {
```

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+		// which breaks `@wordpress/jest-console` assertions.
+
+		// eslint-disable-next-line no-console
+		console.error = jest.fn();
+
+		// Act.
+		const { result } = renderHook( () => useSettings() );
+
+		// Assert.
+		expect( result.error?.message ).toEqual( 'The `useSettings()` hook must be used within an `<SettingsContextProvider />`' );
```
Comment (rotemee) [inline packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:36]:
```suggestion
		expect( result.error?.message ).toEqual( 'The `useSettings()` hook must be used within an `<SettingsProvider />`' );
```

**Review by rotemee — CHANGES_REQUESTED** [review]

File: packages/packages/editor/src/init.tsx (hunk 10 above)
Comment (StyleShit) [reply packages/packages/editor/src/init.tsx:14]: Done

File: tests/phpunit/elementor/core/editor/test-editor-loader.php (hunk 11 above)
Comment (StyleShit) [reply tests/phpunit/elementor/core/editor/test-editor-loader.php:295]: Done

File: tests/phpunit/elementor/core/editor/test-editor-loader.php (hunk 12 above)
Comment (StyleShit) [reply tests/phpunit/elementor/core/editor/test-editor-loader.php:165]: Done

File: packages/packages/editor/src/contexts/settings-context.tsx (hunk 13 above)
Comment (StyleShit) [reply packages/packages/editor/src/contexts/settings-context.tsx:11]: Done

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx (hunk 14 above)
Comment (StyleShit) [reply packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:12]: Done

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx (hunk 15 above)
Comment (StyleShit) [reply packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:24]: Dpne

File: packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx (hunk 16 above)
Comment (StyleShit) [reply packages/packages/editor/src/contexts/__tests__/settings-context.test.tsx:36]: Done

**Review by rotemee — APPROVED** [review]

**Pull request merged**